integrations/nifi/nifi-nar-bundles/nifi-standard-services/nifi-spark-service-bundle/nifi-spark-service/src/test/java/com/thinkbiganalytics/nifi/v2/spark/SparkJobserverServiceIT.java

```diff
@@ -0,0 +1,251 @@
+        String contextName = "testDuplicateContextCreation";
+
+        CreateSparkContext createSparkContext1 = new CreateSparkContext(contextName, numExecutors, memPerNode, numCPUCores, sparkContextType, contextTimeout, async, sparkJobserverService);
+        Thread thread1 = new Thread(createSparkContext1);
+
+        CreateSparkContext createSparkContext2 = new CreateSparkContext(contextName, numExecutors, memPerNode, numCPUCores, sparkContextType, contextTimeout, async, sparkJobserverService);
+        Thread thread2 = new Thread(createSparkContext2);
+
+        thread1.start();
+        thread2.start();
+
+        thread1.join();
+        thread2.join();
+
+        boolean contextExists = sparkJobserverService.checkIfContextExists(contextName);
+
+        Assert.assertTrue(contextExists);
+
+        sparkJobserverService.deleteContext(contextName);
+    }
+
+    /**
+     * Verify creating a Spark Context which times out and gets deleted
+     */
+    @Test
+    @Ignore
+    public void testContextTimeout() throws Exception {
+        // Test creating context
+        String contextName = "testContextTimeout";
+        int contextTimeout = 1;
+
+        boolean created = sparkJobserverService.createContext(contextName, numExecutors, memPerNode, numCPUCores, sparkContextType, contextTimeout, async);
+        Thread.sleep(15000);
+
+        boolean contextExists = sparkJobserverService.checkIfContextExists(contextName);
+        Assert.assertFalse(contextExists);
+    }
+
+    /**
+     * Verify creating two Spark Contexts
+     */
+    @Test
+    @Ignore
+    public void testMultipleContextCreation() throws Exception {
+        // Test creating context
+        String contextOneName = "testMultipleContextCreationOne";
+        String contextTwoName = "testMultipleContextCreationTwo";
+
+        CreateSparkContext createSparkContext1 = new CreateSparkContext(contextOneName, numExecutors, memPerNode, numCPUCores, sparkContextType, contextTimeout, async, sparkJobserverService);
+        Thread thread1 = new Thread(createSparkContext1);
+
+        CreateSparkContext createSparkContext2 = new CreateSparkContext(contextTwoName, numExecutors, memPerNode, numCPUCores, sparkContextType, contextTimeout, async, sparkJobserverService);
+        Thread thread2 = new Thread(createSparkContext2);
+
+        thread1.start();
+        thread2.start();
+
+        thread1.join();
+        thread2.join();
+
+        boolean contextOneExists = sparkJobserverService.checkIfContextExists(contextOneName);
+        boolean contextTwoExists = sparkJobserverService.checkIfContextExists(contextTwoName);
+
+        sparkJobserverService.deleteContext(contextOneName);
+        sparkJobserverService.deleteContext(contextTwoName);
+
+        Assert.assertTrue(contextOneExists && contextTwoExists);
+    }
+
+    /**
+     * Verify executing Spark Context Job
+     */
+    @Test
+    @Ignore
+    public void testExecuteSparkContextJob() throws Exception {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Fix or remove this skipped unit test [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1607)